### PR TITLE
Clean up docs on public chan_utils methods to be more useful

### PR DESCRIPTION
**Pull request by TheBlueMatt**

### CODE REVIEW

**Comment by devrandom** [comment]
We are actually using these for test purposes, but I agree it doesn't make a lot of sense to expose and we can copy the small amount of code to our repo.

File: lightning/src/ln/chan_utils.rs [hunk 1]
```diff
@@ -525,7 +525,11 @@ pub fn make_funding_redeemscript(broadcaster: &PublicKey, countersignatory: &Pub
 	}.push_opcode(opcodes::all::OP_PUSHNUM_2).push_opcode(opcodes::all::OP_CHECKMULTISIG).into_script()
 }
 
-/// panics if htlc.transaction_output_index.is_none()!
+/// Builds an unsigned HTLC-Success or HTLC-Timeout transaction from the given channel and HTLC
```
Comment (valentinewallace) [inline lightning/src/ln/chan_utils.rs:528]: I think `prev_hash` would be clearer if renamed to `commitment_tx`

**Review by valentinewallace — COMMENTED** [review]
Maybe both could use something about when/why you'd want to call them?

**Review by valentinewallace — APPROVED** [review]

**Review by jkczyz — APPROVED** [review]
Looks good after squash

**Comment by TheBlueMatt** [comment]
Squashed without changes:
```
$ git diff-tree -U1 22914572f 8cdc855f0
$
```

**Pull request merged**